src/node.cc

```diff
@@ -5014,6 +5023,11 @@ int Start(int argc, char** argv) {
       TTDFlagWarning_Cond(s_ttoptReplayUri != nullptr,
           "Must set replay source info when replaying.\n");
   }
+
+  if (!envDoRecord && s_doTTRecord) {
```

Comment:
You're right about that, but if envDoRecord is already true, then the environment variable doesn't need to be set anyway.  I just did this since there's no point in setting the environment variable if we already processed it and determined it to be enabled.